### ISSUE TITLE
Short backtrace.

### ISSUE THREAD

**Issue by vi** [rust-lang-nursery/failure#256]
With `RUST_BACKTRACE=1`, `Debug` outputs too much - entire screenful of a backtrace (similar to `RUST_BACKTRACE=full` when panic)

Can it instead output short backtrace (not going into backtrace capturing or failure's internal details, not continuing after "main") unless it's `RUST_BACKTRACE=full`?

**Comment by mitsuhiko** [comment]
Yeah. I think rendering a shortened backtrace would be useful. Accepting patches but I might also do this myself when I find time.
